File: csharp/Original/PublishingHouse/Books/Entities/Publisher.cs
using PublishingHouse.Core.ValueObjects;

namespace PublishingHouse.Books.Entities;

public record Publisher(PublisherId Id, PublisherName Name);
public record PublisherId(Guid Value) : NonEmptyGuid(Value);
public record PublisherName(string Value) : NonEmptyString(Value);

File: node.js/src/original/domain/books/entities/publisher.ts
import { DeepReadonly, NonEmptyString, NonEmptyUuid } from '#core/typing';

export type PublisherId = NonEmptyUuid<'PublisherId'>;
export type PublisherName = NonEmptyString<'PublisherName'>;

export type Publisher = DeepReadonly<{ Id: PublisherId; Name: PublisherName }>;
